packages/core/src/utils/JsonTransformer.ts

```diff
@@ -1,15 +1,40 @@
+import type { Validate } from 'class-validator'
+
 import { instanceToPlain, plainToInstance, instanceToInstance } from 'class-transformer'
 
+import { ClassValidationError } from '../error/ClassValidationError'
+
+import { MessageValidator } from './MessageValidator'
+
+interface Validate {
+  validate?: boolean
+}
+
 export class JsonTransformer {
   public static toJSON<T>(classInstance: T) {
     return instanceToPlain(classInstance, {
       exposeDefaultValues: true,
     })
   }
 
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  public static fromJSON<T>(json: any, Class: { new (...args: any[]): T }): T {
-    return plainToInstance(Class, json, { exposeDefaultValues: true })
+  public static fromJSON<T>(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    json: any,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    cls: { new (...args: any[]): T },
+    { validate }: Validate = { validate: true }
```

Comment:
validate should be set to true on the left side of the equation

See the output of this script for why we want this (https://www.typescriptlang.org/play?#code/PTAEHUFMBsGMHsC2lQBd5oBYoCoE8AHSAZVgCcBLA1UABWgEM8BzM+AVwDsATAGiwoBnUENANQAd0gAjQRVSQAUCEmYKsTKGYUAbpGF4OY0BoadYKdJMoL+gzAzIoz3UNEiPOofEVKVqAHSKymAAmkYI7NCuqGqcANag8ABmIjQUXrFOKBJMggBcISGgoAC0oACCbvCwDKgU8JkY7p7ehCTkVDQS2E6gnPCxGcwmZqDSTgzxxWWVoASMFmgYkAAeRJTInN3ymj4d-jSCeNsMq-wuoPaOltigAKoASgAywhK7SbGQZIIz5VWCFzSeCrZagNYbChbHaxUDcCjJZLfSDbExIAgUdxkUBIursJzCFJtXydaiKDIKMjJBhLABqDGgFG4dRQAG9FCUSjoGUyWQB+fLjeDwFqcRQAX0UCE4gho7mSNAAvKAABSs0DcxnMhSgZWoMjsFDiwX0rUs3WgVnigCUuoAfGiZSLIAFoPBmCqAETy1Ce-ia3kKa3BH0q4Ohq3hyAKtUannayCC6nQQRG4NSxqy0CUZiYJWq9UBhOgY2gU2BlDKwvxlmC-WGm32x2CZ2u91enN5v1xs1B4Kd1BhxQDtU24cUXOD6u9xM4hmpkvBlQ4ZoeMheRDwPoMYHsGhfNxmL5kd6p-jMSBHVCOBSuaR4bOQTc6YZYfQoBCIaHCS4HiRb6IRAMDhsQWJhWA4HgAiAA)

Reply:
@blu3beri have a look at the link.